### PR TITLE
Split the docs for "Standard Library" into "Base" and "Standard Library"

### PR DESCRIPTION
**Pull request by KristofferC**
The docs right now is split into Manual and Standard Library (and Developer Docs). However, we are now more and more differentiating between what is part of "Base" and what is the "Standard Library". Therefore, I think it makes sense to split the header that is currently called "Standard Library" into two parts; one called "Base" (corresponding to functionality in the `base` foler) and one called "Standard Library" (which corresponds to the modules in `stdlib`).

WIP because the introduction to Base needs to be rewritten to not call it the Standard Library, and because there should probably be an introduction corresponding to the Standard Library header.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
💯

**Comment by Sacha0** [comment]
This pull request brings the discussion in https://github.com/JuliaLang/julia/pull/23876#issuecomment-333163762 to mind. Migrating to `stdpkg`/"standard packages" would be a nice way to avoid ambiguity / confusion and maintain clear separation. Best!

**Comment by KristofferC** [comment]
Personally, I find the Base / Standard Library distinction to work quite well. Standard packages sounds a bit like an oxymoron to me.

**Comment by KristofferC** [comment]
Alright, updated this. For those places in the Base documentation that were referring to Base as "the standard library", I changed this to "Julia Base". Any other suggestions?

There are a few places in the Manual section (Date and Datetime, Profiling) that should probably be moved completely to their stdlib documentation, but this can be done later.

This is no longer WIP so please review.

**Review by fredrikekre — APPROVED** [review]
Very nice!

**Comment by fredrikekre** [comment]
Timeout on one Appveyor worker.

File: doc/make.jl [hunk 1]
```diff
@@ -162,13 +139,13 @@ const PAGES = [
...
-      Dates, IterativeEigensolvers, Unicode, Distributed, Printf
+for stdlib in STDLIB_DOCS
+    @eval using $(stdlib.stdlib)
+end
 
 makedocs(
     build     = joinpath(pwd(), "_build/html/en"),
-    modules   = [Base, Core, BuildSysImg, DelimitedFiles, Test, Mmap, SharedArrays, Profile,
-                 Base64, FileWatching, Dates, IterativeEigensolvers, Unicode, Distributed, Printf],
+    modules   = [Base, Core, BuildSysImg, [Module(stdlib.stdlib) for stdlib in STDLIB_DOCS]...],
```
Comment (JeffBezanson) [inline doc/make.jl]: Does this work? I thought `Module(x)` made a new empty module.

Great change though :+1:

Comment (KristofferC) [reply doc/make.jl]: > I thought `Module(x)` made a new empty module.

Well, clearly you haven't been programming Julia long enough...

It works in the sense that the docs render correctly as far as I can see.

Comment (JeffBezanson) [reply doc/make.jl]: That's weird; something is up. `stdlib.stdlib` is a Symbol right?

```
julia> using Unicode

julia> Module(:Unicode) === Unicode
false
```

Comment (KristofferC) [reply doc/make.jl]: Ok, it is wrong but the reason it still work, I think, is that none of the stdlib modules have any submodules, which is what Documenter uses this information. Will correct.

**Pull request merged**